WordPress/openverse-api
# Register audio and image models consistently in admin panel

## Fixes


Fixes #919 by @krysal 

## Description


This PR
- ensures that images and audio have all reporting models registered in the admin panel
- adds checks to ensure non-existent identifiers cannot be reported
- distributes concerns from the central `_Report` models to the `Deleted_` and `Mature_` models
- ensures that the changes to mature status are synced to Elasticsearch
- surface `mature` field in the details endpoint for WordPress/openverse-frontend#1822
- limits admin functionality to encourage correct usage of reporting flows
- adds documentation where it was lacking
- implements general refactors and makes code quality improvements

## Testing Instructions

0. Check out the PR and run all services.
1. Visit the Django admin panel.
2. See new models present.

## Checklist


[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
